homework/list.js

```diff
@@ -1,79 +1,108 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
-
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
+  /**
+   * Creates an instance of ListManager.
+   *
+   * @constructor
+   * @this {ListManager}
+   */
+  var ListManager = function() {
+    this._listNoteContent = [];
+    this._wrapper = document.querySelector('#note-list-wrapper');
   }
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+  ListManager.prototype = {
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
+    /**
+     * Binding the 'click' event to the '#note-list-wrapper' node and
+     * fetch the data from server then draw it.
+     * This method should be called only once to initialize the ListManager.
+     *
+     * @this {ListManager}
+     */
+    start() {
+      this.fetchList().then((function(data){
```

Comment:
Arrow function is more expressive than using `bind` API. See:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions